components/atom/switch/src/index.js

```diff
@@ -0,0 +1,160 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import {ToggleSwitchTypeRender} from './SwitchType/toggle'
+import {SingleSwitchTypeRender} from './SwitchType/single'
+
+export const BASE_CLASS = 'sui-AtomSwitch'
+
+export const SIZES = {
+  DEFAULT: 'default',
+  LARGE: 'large'
+}
+
+export const TYPES = {
+  TOGGLE: 'toggle',
+  SELECT: 'select',
+  SINGLE: 'single'
+}
+
+class AtomSwitch extends Component {
+  state = {
+    toggle: false,
+    isFocus: false
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.executeIfEnabledFocusSwitch = this.executeIfEnabled(this.focusSwitch)
+    this.executeIfEnabledFocusOutSwitch = this.executeIfEnabled(this.focusOutSwitch)
+    this.executeIfEnabledToggleSwitch = this.executeIfEnabled(this.toggleSwitch)
+    this.executeIfEnabledActivateSwitch = this.executeIfEnabled(this.activateToggle)
+    this.executeIfEnabledDeactivateSwitch = this.executeIfEnabled(this.deactivateToggle)
+  }
+
+  componentDidMount () {
+    this.keyBindingsInstance = this.keyBindings.bind(this)
+    window.addEventListener('keydown', this.keyBindingsInstance, true)
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('keydown', this.keyBindingsInstance, true)
+    this.keyBindingsInstance = null
+  }
+
+  keyBindings = (event) => {
+    const {isFocus} = this.state
+    if (!isFocus || event.defaultPrevented) {
+      return
+    }
+    if (event.key === 'Enter' || event.key === ' ') {
+      this.toggleSwitch()
+      event.preventDefault()
+    }
+  }
+
+  toggleSwitch = () => {
+    const {toggle} = this.state
+    const {onToggle} = this.props
+    onToggle(!toggle)
+    this.setState({toggle: !toggle})
```

Comment:
I think it could be dangerous to call the callback before changing the state. In case you agree `this.setState({toggle: !toggle}, () => onToggle(!toggle))` should do the job